core/src/main/scala/org/apache/spark/status/api/v1/AllStagesResource.scala

```diff
@@ -237,7 +237,8 @@ private[v1] object AllStagesResource {
   }
 
   def convertAccumulableInfo(acc: InternalAccumulableInfo): AccumulableInfo = {
-    new AccumulableInfo(acc.id, acc.name, acc.update, acc.value)
+    new AccumulableInfo(
+      acc.id, acc.name, acc.update.map(_.toString), acc.value.map(_.toString).orNull)
```

Comment:
This was kind of confusing on first glance until I rembered that we have the weird UI AccumulableInfo and the other version which is used elsewhere and which has been renamed to `InternalAccumulableInfo` here.